addons/mail/models/mail_message.py

```diff
@@ -326,8 +320,13 @@ def _message_read_dict_postprocess(self, messages, message_tree):
                                 if partner.id in partner_tree]
 
             customer_email_data = []
-            for notification in message.notification_ids.filtered(lambda notif: notif.res_partner_id.partner_share and notif.res_partner_id.active):
-                customer_email_data.append((partner_tree[notification.res_partner_id.id][0], partner_tree[notification.res_partner_id.id][1], notification.email_status))
+
+            active_test = self.env.context.get('notification_active_test', True)
```

Comment:
I don't like using the context. I think it would be better to add an explicit parameter somewhere in the routing to explicit ask for already-read messages.

Reply:
Hello @tde-banana-odoo ,
We are passing context value based on channels, and we need this value in child method of 'message_fetch' i.e. '_message_read_dict_postprocess', in this method we need to search all the notifications based on the channel_id's condition (for history its true and for inbox its false), but the reason behind not passing it as a parameter is, if we pass it as a parameter we have to pass it to the three methods i.e. 'message_fetch', 'message_format' and '_message_read_dict_postprocess', which is not convinient as per my knowledge, please give your valuable inputs on this.
Thanks.